### PR TITLE
Add mock data to provenance graph

### PR DESCRIPTION
**Pull request by emilykuang**
**What** 

- Added mock data to provenance graph UI skeleton 
- **Update:** Added ProvenanceGraphInterface and change the Condensed | Expanded buttons to a toggle that shows the active layout

**How**

- Added `ProvenanceRenderer` to render operations as rectangles and model, parameters, and variables as ellipses

**Testing**

- Open Simulation View with one model and toggle the condensed and expanded buttons to see the single model workflow 
- Select two models and open simulation view to see the multi-model workflow 

**Screen Recording**

https://user-images.githubusercontent.com/23723378/129083716-540904f9-9ec5-4fb9-9291-9d5eb932e8b2.mov

**Updated Recording with Toggle**
https://user-images.githubusercontent.com/23723378/129400113-40098179-c7aa-4a2e-9d6d-fe413d76be0c.mov

### CODE REVIEW

File: client/src/views/Simulation/components/ProvenanceGraph/ProvenanceGraph.vue [hunk 1]
```diff
@@ -63,6 +62,11 @@
   z-index: var(--z-index--side-panel);
 }
 
+.provenance-graph-data {
+  height: 24vh;
```
Comment (YohannParis) [inline client/src/views/Simulation/components/ProvenanceGraph/ProvenanceGraph.vue:66]: This is worth adding to `styles/ui.css` as a CSS custom property.

File: client/src/views/Simulation/components/ProvenanceGraph/ProvenanceGraphData.vue [hunk 2]
```diff
@@ -1,32 +1,67 @@
 <template>
-  <section class="provenance-graph-data">
-    <h6>{{header}}</h6>
-  </section>
+  <div class="provenance-graph-data" ref="graph" />
```
Comment (YohannParis) [inline client/src/views/Simulation/components/ProvenanceGraph/ProvenanceGraphData.vue:2]: 😢 we lost the `section` for a sad `div`.

File: client/src/views/Simulation/components/ProvenanceGraph/ProvenanceGraphData.vue [hunk 3]
```diff
@@ -1,32 +1,67 @@
...
+  import { Prop, Watch } from 'vue-property-decorator';
+
+  import { expandCollapse } from 'svg-flowgraph';
+
+  import { GraphInterface } from '@/types/typesGraphs';
+
+  import ProvenanceRenderer from '@/graphs/svg/renderers/ProvenanceRenderer';
+  import ELKAdapter from '@/graphs/svg/elk/adapter';
+
+  const DEFAULT_RENDERING_OPTIONS = {
```
Comment (YohannParis) [inline client/src/views/Simulation/components/ProvenanceGraph/ProvenanceGraphData.vue:17]: instead you should import the one from the `EpiRenderer`:
```JS
import { DEFAULT_RENDERING_OPTIONS } from '@/graphs/svg/renderers/EpiRenderer';
```

**Review by YohannParis — APPROVED** [review]
Looks really neat, while following closely the codebase style 👍

File: client/src/views/Simulation/components/ProvenanceGraph/ProvenanceGraphData.vue (hunk 3 above)
Comment (emilykuang) [reply client/src/views/Simulation/components/ProvenanceGraph/ProvenanceGraphData.vue]: Yeah I think they can look alike, I've imported from `EpiRenderer`

File: client/src/views/Simulation/components/ProvenanceGraph/ProvenanceGraphData.vue (hunk 2 above)
Comment (emilykuang) [reply client/src/views/Simulation/components/ProvenanceGraph/ProvenanceGraphData.vue]: haha oops I changed it back to a section

**Review by YohannParis — APPROVED** [review]
Well done!

File: client/src/graphs/svg/renderers/ProvenanceRenderer.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+      .style('stroke', 'none');
+  }
+
+  renderNode (nodeSelection: d3.Selection<any, any, any, any>): void {
+    nodeSelection.each(function () {
+      const selection = d3.select(this);
+      const datum = selection.datum();
+
+      // Operations are displayed as rectangles while models, parameters, and variables are displayed as ellipses
+      if (!(datum as any).data.role?.includes('Operation')) {
```
Comment (RosaRomeroGomez) [inline client/src/graphs/svg/renderers/ProvenanceRenderer.ts:80]: I think you are actually doing the opposite of what you say in the comment, I think you need to remove the logical not (!) and that should do it.

File: client/src/views/Simulation/Simulation.vue [hunk 5]
```diff
@@ -269,6 +277,24 @@
       });
     }
 
+    get provenanceData (): Graph.GraphInterface {
+      // If one model is selected, show the provenance graphs for the single model workflow
+      if (this.selectedModels.length === 1) {
+        if (this.selectedProvenanceLayout === ProvenanceLayoutInterfaceType.condensed) {
+          return ProvenanceData.SINGLE_MODEL_CONDENSED;
```
Comment (RosaRomeroGomez) [inline client/src/views/Simulation/Simulation.vue:284]: Maybe you can simplify these conditions by using a ternary one:
```

if (this.selectedModels.length === 1) { 
  return this.selectedProvenanceLayout === ProvenanceLayoutInterfaceType.condensed ? ProvenanceData.SINGLE_MODEL_CONDENSED : ProvenanceData.SINGLE_MODEL_EXPANDED
} else ....
```

File: client/src/views/Simulation/Simulation.vue [hunk 6]
```diff
@@ -269,6 +277,24 @@
...
+        } else {
+          return ProvenanceData.SINGLE_MODEL_EXPANDED;
+        }
+      } else {
+      // If multiple models are selected, show the provenance graph for the multiple model workflow
+        if (this.selectedProvenanceLayout === ProvenanceLayoutInterfaceType.condensed) {
```
Comment (RosaRomeroGomez) [inline client/src/views/Simulation/Simulation.vue:290]: The same here.

File: client/src/graphs/svg/renderers/ProvenanceRenderer.ts (hunk 4 above)
Comment (emilykuang) [reply client/src/graphs/svg/renderers/ProvenanceRenderer.ts]: Thanks for catching this!

**Review by RosaRomeroGomez — APPROVED** [review]
Thanks!

**Pull request merged**